### ISSUE TITLE
Dynamic Imports and React Create App

### ISSUE THREAD

**Issue by tim-field** [cypress-io/cypress#2457]
I'm using react create app, which supports `dyanmicImports` out of the box. 

However when I run cypress and try to import some code from my main app into my tests  ( and maybe this is something I shouldn't be doing :thinking: ) I get;

```
Support for the experimental syntax 'dynamicImport' isn't currently enabled (28:17):
```

My routes code uses dynamic imports.  And for the sake of an example, say I'd like to use a method from this file (`getViewRoute`) in my tests

```JS
import {getViewRoute} from '../../../src/routes';

it("Page has loaded", () => {
      const url = getViewRoute("foo").url({id: "bar"})
      cy.visit(url)
      cy.get(`[data-test=ListObjectPage h1`)
  })
```

I've tried manually installing babel and the required models to my project.

```JSON
{
  "presets": ["react"],
  "plugins": [
    "@babel/plugin-syntax-dynamic-import",
    "@babel/plugin-transform-modules-commonjs",
    "@babel/plugin-proposal-object-rest-spread"
  ]
}
```

### Current behavior:

Dynamic imports work in my react app, but they don't when testing with cypress.

### Desired behavior:

Dynamic imports should work in cypress as they do in the app

### Steps to reproduce:

Setup a project with react create app,
use the dynamic import syntax
test code that relies on this.

### Versions

"cypress": "^3.1.0",
chrome 69

**Comment by chrisbreiding** [comment]
You're on the right track. You need to configure the preprocessing of your spec files to enable dynamic imports through the [`file:preprocessor` event in the plugins file](https://docs.cypress.io/guides/tooling/plugins-guide.html#Preprocessors). 

The easiest way is probably to use the [webpack preprocessor](https://github.com/cypress-io/cypress-webpack-preprocessor). Install that an take a look at the [section on modifying default options](https://github.com/cypress-io/cypress-webpack-preprocessor#modifying-default-options) for how to add the babel plugin(s) you need.

**Comment by tim-field** [comment]
Thanks, I'm closer now.  Currently tripping up with 

```
Module build failed (from ./node_modules/@cypress/webpack-preprocessor/node_modules/babel-loader/lib/index.js):
Error: Requires Babel "^7.0.0-0", but was loaded with "6.26.0". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel.
```

I'm using babel 7,  I'm wondering if  [webpack preprocessor](https://github.com/cypress-io/cypress-webpack-preprocessor) requires babel 6 ?

**Comment by chrisbreiding** [comment]
I just released [v3.0.0](https://github.com/cypress-io/cypress-webpack-preprocessor/releases/tag/v3.0.0), which upgrades Babel to v7, so please update to that and let me know if you still have any issues.

**Comment by tim-field** [comment]
Thanks!
